sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/constraints.scala

```diff
@@ -92,6 +98,11 @@ trait TableConstraint {
       )
     }
   }
+
+  override def nullable: Boolean = true
+
+  override def dataType: DataType =
+    throw new SparkUnsupportedOperationException("CONSTRAINT_DOES_NOT_HAVE_DATA_TYPE")
```

Comment:
we should throw `UnresolvedException`, to follow other similar expressions like `MergeAction`